Code/Source/RobotImporter/URDF/PrefabMakerUtils.cpp

```diff
@@ -0,0 +1,134 @@
+        {
+            AZ_Error("AddVisuals", false, "Could not find model asset for %s", modelPath.c_str());
+            return {};
+        }
+
+        auto assetPath = AZ::IO::Path(assetInfo.m_relativePath).ReplaceExtension("azmodel");
+        AZStd::to_lower(assetPath.Native().begin(), assetPath.Native().end());
+
+        return assetPath;
+    }
+
+    void SetEntityTransform(const urdf::Pose& origin, AZ::EntityId entityId)
+    {
+        urdf::Vector3 urdfPosition = origin.position;
+        urdf::Rotation urdfRotation = origin.rotation;
+        AZ::Quaternion azRotation = URDF::TypeConversions::ConvertQuaternion(urdfRotation);
+        AZ::Vector3 azPosition = URDF::TypeConversions::ConvertVector3(urdfPosition);
+        AZ::Transform tf(azPosition, azRotation, 1.0f);
+
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        auto* transformInterface = entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+
+        if (!transformInterface)
+        {
+            AZ_Error("SetEntityTransform", false, "Missing Transform component!");
+            return;
+        }
+        // Alternative - but requires an active component. We would activate/deactivate a lot. Or - do it in a second pass.
+        // AZ::TransformBus::Event(entityId, &AZ::TransformBus::Events::SetLocalTM, transformForChild);
+
+        transformInterface->SetLocalTM(tf);
+    }
+
+    AzToolsFramework::Prefab::PrefabEntityResult CreateEntity(AZ::EntityId parentEntityId, const AZStd::string& name)
+    {
+        auto prefabInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabPublicInterface>::Get();
+        auto createEntityResult = prefabInterface->CreateEntity(parentEntityId, AZ::Vector3());
+        if (!createEntityResult.IsSuccess())
+        {
+            return createEntityResult;
+        }
+
+        AZ::EntityId entityId = createEntityResult.GetValue();
+        if (!entityId.IsValid())
+        { // Verify that a valid entity is created.
+            return AZ::Failure(AZStd::string("Invalid id for created entity"));
+        }
+
+        AZ_TracePrintf("CreateEntity", "Processing entity id:%s with name:%s", entityId.ToString().c_str(), name.c_str());
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        AZStd::string entityName(name.c_str());
+        entity->SetName(entityName);
+        entity->Deactivate();
+        AddRequiredComponentsToEntity(entityId);
+        return createEntityResult;
+    }
+
+    AzToolsFramework::Prefab::PrefabOperationResult RemoveEntityWithDescendants(AZ::EntityId parentEntityId)
+    {
+        auto prefabInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabPublicInterface>::Get();
+        return prefabInterface->DeleteEntitiesAndAllDescendantsInInstance({ parentEntityId });
+    }
+
+    void AddRequiredComponentsToEntity(AZ::EntityId entityId)
+    {
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
```

Comment:
Looks like a good place to protect against an invalid ```entityId```. (```*entity``` below may cause a NPE)